File: dev-docs/snippets/auth/magic-link-request.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Email a magic sign-in link to the user.
export async function magicLinkRequest(client: JsBaoClient, email: string) {
  // #region example
  const { success } = await client.magicLinkRequest(email);
  // #endregion example
  return success;
}

File: dev-docs/snippets/auth/magic-link-request.swift
import JsBaoClient

// Email a magic sign-in link to the user via the typed `client.auth`
// namespace. Swift requires an explicit `redirectUri` and returns a typed
// `MagicLinkRequestResult` (`success`).
func magicLinkRequest(
  client: JsBaoClient,
  email: String,
  redirectUri: String
) async throws {
  // #region example
  let result = try await client.auth.magicLinkRequest(
    email: email,
    redirectUri: redirectUri
  )
  let success = result.success
  // #endregion example
  _ = success
}
